drivers/include/at.h

```diff
@@ -0,0 +1,202 @@
+ *
+ * This module provides functions to interact with devices using AT commands.
+ *
+ * Most functions compare the bytes echoed by the device with what they
+ * intended to send, and bail out if there's no match.
+ *
+ * Furthermore, the library tries to copy with difficulties regarding different
+ * line endings. It usually sends "<command><CR>", but expects
+ * "<command>\LF\CR" as echo.
+ *
+ * As a debugging aid, when compiled with "-DAT_PRINT_INCOMING=1", every input
+ * byte gets printed.
+ * @{
+ *
+ * @file
+ *
+ * @brief       AT (Hayes) library interface
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ */
+
+#ifndef AT_H
+#define AT_H
+
+#include <stdint.h>
+
+#include "isrpipe.h"
+#include "periph/uart.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef AT_END_OF_LINE
+/** End of line character to send after the AT command */
+#define AT_END_OF_LINE "\r"
+#endif
+
+/**
+ * @brief AT device structure
+ */
+typedef struct {
+    isrpipe_t isrpipe;      /**< isrpipe used for getting data from uart */
+    uart_t uart;            /**< UART device where the AT device is attached */
+} at_dev_t;
+
+
+/**
+ * @brief   Initialize AT device struct
+ *
+ * @param[in]   dev         struct to initialize
+ * @param[in]   uart        UART the device is connected to
+ * @param[in]   baudrate    baudrate of the device
+ * @param[in]   buf         input buffer
+ * @param[in]   bufsize     size of @p buf
+ *
+ * @returns     0 on success
+ * @returns     <0 otherwise
+ */
+int at_dev_init(at_dev_t *dev, uart_t uart, uint32_t baudrate, char *buf, size_t bufsize);
+
+/**
+ * @brief   simple command helper
+ *
+ * This function sends an AT command to the device and waits for "OK".
+ *
+ * @param[in]   dev     device to operate on
+ * @param[in]   command command string to send
+ * @param[in]   timeout timeout (in usec)
+ *
+ * @returns     0 when device answers "OK"
+ * @returns     <0 otherwise
+ */
+int at_send_cmd_wait_ok(at_dev_t *dev, const char *command, uint32_t timeout);
+
+/**
+ * @brief   send AT command, wait for a prompt
+ *
+ * This function will send the supplied @p command, then wait for the prompt (>)
+ * character and return
+ *
+ * @param[in]   dev     device to operate on
+ * @param[in]   command command string to send
+ * @param[in]   timeout timeout (in usec)
+ *
+ * @return      0 when prompt is received
+ * @return      <0 otherwise
+ */
+int at_send_cmd_wait_prompt(at_dev_t *dev, const char *command, uint32_t timeout);
+
+/**
+ * @brief   send AT command, wait for response
+ *
+ * This function will send the supplied @p command, then wait and return one
+ * line of response.
+ *
+ * A possible empty line will be skipped.
+ *
+ * @param[in]   dev         device to operate on
+ * @param[in]   command     command to send
+ * @param[out]  resp_buf    buffer for storing response
+ * @param[in]   len         len of @p buffer
+ * @param[in]   timeout     timeout (in usec)
+ *
+ * @returns     lenght of response on success
```

Comment:
before I forget, there's a typo: s/lenght/length/. I saw it in other places.

Reply:
fixed two times int at.h.